src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -1,126 +1,188 @@
       val dao = AgoraDao.createAgoraDao(entityTypes)
       val entityWithId = dao.findSingle(agoraEntity.namespace.get, agoraEntity.name.get, agoraEntity.snapshotId.get)
-      val configurations = dao.findConfigurations(entityWithId.id.get)
-
-      configurations.foreach {config => AgoraEntityPermissionsClient.deleteAllPermissions(config)}
+      dao.findConfigurations(entityWithId.id.get)
+    } else {
+      Seq()
     }
 
-    AgoraEntityPermissionsClient.deleteAllPermissions(agoraEntity)
+    permissionsDataSource.inTransaction { db =>
+      checkNamespacePermission(db, agoraEntity, username, AgoraPermissions(Redact), checkAdmin = true) { //admins can redact anything
+        DBIO.sequence(configurations map { config => db.aePerms.deleteAllPermissions(config) }) andThen
+          db.aePerms.deleteAllPermissions(agoraEntity)
+      }
+    }
   }
 
   def find(agoraSearch: AgoraEntity,
            agoraProjection: Option[AgoraEntityProjection],
            entityTypes: Seq[AgoraEntityType.EntityType],
-           username: String): Seq[AgoraEntity] = {
+           username: String): Future[Seq[AgoraEntity]] = {
 
     val entities = AgoraDao.createAgoraDao(entityTypes)
       .find(agoraSearch, agoraProjection)
       .map(entity => entity.addUrl().removeIds())
 
-    AgoraEntityPermissionsClient.filterEntityByRead(entities, username)
+    permissionsDataSource.inTransaction { db =>
+      db.aePerms.filterEntityByRead(entities, username)
+    }
   }
 
   def findSingle(namespace: String,
                  name: String,
                  snapshotId: Int,
                  entityTypes: Seq[AgoraEntityType.EntityType],
-                 username: String): AgoraEntity = {
+                 username: String): Future[AgoraEntity] = {
     val foundEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId)
-    val entities = AgoraEntityPermissionsClient.filterEntityByRead(Seq(foundEntity), username)
-    entities match {
-      case Seq(ae: AgoraEntity) => ae.addUrl().removeIds().addManagers(AgoraEntityPermissionsClient.listOwners(foundEntity))
-      case _ => throw new AgoraEntityNotFoundException(foundEntity)
+
+    permissionsDataSource.inTransaction { db =>
+      db.aePerms.filterEntityByRead(Seq(foundEntity), username) flatMap {
+        case Seq(ae: AgoraEntity) => db.aePerms.listOwners(foundEntity) map { owners => ae.addUrl().removeIds().addManagers(owners) }
+        case _ => DBIO.failed(AgoraEntityNotFoundException(foundEntity))
+      }
     }
   }
 
   def findSingle(entity: AgoraEntity,
                  entityTypes: Seq[AgoraEntityType.EntityType],
-                 username: String): AgoraEntity = {
+                 username: String): Future[AgoraEntity] = {
     findSingle(entity.namespace.get, entity.name.get, entity.snapshotId.get, entityTypes, username)
   }
 
-  private def validatePayload(agoraEntity: AgoraEntity, username: String): Unit = {
```

Comment:
this moved [to the top](https://github.com/broadinstitute/agora/pull/182/files#diff-1768f9d51119592601a48548da8f9904R64) because i turned it into a check function, rather than something that just barfs an exception into your lap